### PR TITLE
Fix `babel-preset-env`.

### PR DESCRIPTION
**Pull request by tech4him1**
**- Summary**

We added `babel-preset-env` before, but I was using the docs for the beta version instead of the released version, so it was actually still transpiling everything instead of just the stuff we needed.

**- Test plan**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: webpack.prod.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const path = require('path');
 const webpack = require('webpack');
 const merge = require('webpack-merge');
 const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const UglifyJSPlugin = require('uglifyjs-webpack-plugin');
```
Comment (erquhart) [inline webpack.prod.js]: Should be `UglifyJsPlugin` (lowercase "s").

Comment (tech4him1) [reply webpack.prod.js]: Yup, already figured that out, thank you. This is throwing some really weird errors about components not being mounted (after this line is fixed), please check it out if you get a chance.

**Comment by tech4him1** [comment]
OK, here is the problem with React Hot Loader: https://github.com/gaearon/react-hot-loader/issues/695.

**Comment by tech4him1** [comment]
React Hot Loader was removed in 1.0, so this should be good to merge now. We'll need to consider this if we ever add React Hot Loader in again, but they are working on fixing the problem, so hopefully it will be resolved by then.

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit 9ffa3b694cb2eb7d218dd7822b3a1cd25001a2b9

https://deploy-preview-765--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit fbdcb3ffe1b946b98d9c39156cb10e40d436726a

https://deploy-preview-765--netlify-cms-www.netlify.com

**Comment by tech4him1** [comment]
@erquhart @Benaiah Rebased on 1.0 and ready for review.

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit d08b8019fc1b37470b6a9cab4dd2b486ace23941

https://deploy-preview-765--cms-demo.netlify.com

**Comment by tech4him1** [comment]
@erquhart @Benaiah Rebased on 1.0.3 and ready for review.

**Comment by tech4him1** [comment]
@erquhart already reviewed this, do you want me to just merge?

**Comment by erquhart** [comment]
Yep, go for it

**Comment by tech4him1** [comment]
Blocking, this is now breaking tests.

**Comment by erquhart** [comment]
Added node 6 as a target, should pass now.

**Comment by tech4him1** [comment]
@erquhart Should we just upgrade our test environment?

**Comment by erquhart** [comment]
Keeping it this way allows contributors with Node 6 to still run the test suite.

**Comment by tech4him1** [comment]
Hmm, Node 6 is still supported, so I guess we could. Just doesn't make very much sense to me to transpile stuff just for our test environment when all the browsers support it.

**Comment by tech4him1** [comment]
@erquhart Hmm, Node 6 is still supported, so I guess we could. Just doesn't make very much sense to me to transpile stuff just for our test environment when all the browsers support it.

**Comment by erquhart** [comment]
That's what I don't know - are we transpiling for node in the build or just on the fly during testing?

**Comment by erquhart** [comment]
It appears this does impact the build. Odd.

Also, it appears we're now in a parallel universe where browsers support pretty much everything natively?

screen shot 2018-01-24 at 12 17 36 pm

Honestly, I'd hate to require Node 7+. Performance matters, but we're not exactly dying in that department. I vote we open a separate issue about it and keep this one focused on using babel-preset-env.

**Comment by tech4him1** [comment]
That works, feel free to merge.

Here is what we are currently transpiling with `babel-preset-env`, and why:
```
Using plugins:
  transform-es2015-destructuring {"edge":"15","node":"6"}
  transform-es2015-for-of {"node":"6"}
  transform-es2015-function-name {"edge":"15","node":"6"}
  transform-exponentiation-operator {"node":"6"}
  transform-async-to-generator {"node":"6"}
  syntax-trailing-function-commas {"node":"6"}
```

**Comment by tech4him1** [comment]
@erquhart We could use `"node": "current"` to transform only the features that need to be for each Node version.
> Support a target option `"node": "current"` to compile for the currently running node version.
> For example, if you are building on Node 6, arrow functions won't be converted, but they will if you build on Node 0.12.

I'm not sure I'd want the build to be different than the test environment, though, any thoughts?

**Comment by tech4him1** [comment]
Hmm, but then if you run a build on an old node version (v4?) it will transpile a whole bunch of stuff.

@erquhart I guess the core question is still whether we want to have the build and test environment transpiling the same things (like you have now), or different?

**Comment by erquhart** [comment]
> We could use "node": "current" to transform only the features that need to be for each Node version. Hmm, but then if you run a build on an old node version (v4?) it will transpile a whole bunch of stuff.

Yeah we don't want to depend on the build environment, I think that makes more sense for actual Node apps.

> I guess the core question is still whether we want to have the build and test environment transpiling the same things (like you have now), or different?

We could maybe run production releases without Node 6 support, but all other commands (including a non-production build command) still support it.

**Comment by tech4him1** [comment]
Do you think there could be any problems having our production env different from our dev one?

**Comment by erquhart** [comment]
Only if there are bugs in the browser implementations of the features we're not polyfilling, which is a really great point. Maybe just for running tests, then.

**Comment by tech4him1** [comment]
You're still having a different environment for your tests vs your build then.

**Comment by erquhart** [comment]
The only difference is in browser implementations of those features vs the babel polyfills running for the tests, so a browser specific bug is the only thing that would slip through, and that's no different from what we're already doing (since our unit tests aren't running against multiple browsers).

**Pull request merged**